plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/psiUtils.kt

```diff
@@ -49,22 +50,43 @@ fun canKeepEqEq(left: PsiExpression, right: PsiExpression?): Boolean {
 fun PsiMember.visibility(
     referenceSearcher: ReferenceSearcher,
     assignNonCodeElements: ((JKFormattingOwner, PsiElement) -> Unit)?
-): JKVisibilityModifierElement =
-    modifierList?.children?.mapNotNull { child ->
+): JKVisibilityModifierElement {
+    if (parentOfType<PsiClass>()?.isInterface == true) {
+        // interface methods are always public
+        return JKVisibilityModifierElement(Visibility.PUBLIC)
+    }
+    val visibility = modifierList?.children?.mapNotNull { child ->
         if (child !is PsiKeyword) return@mapNotNull null
         when (child.text) {
             PsiModifier.PACKAGE_LOCAL -> Visibility.INTERNAL
             PsiModifier.PRIVATE -> Visibility.PRIVATE
             PsiModifier.PROTECTED -> handleProtectedVisibility(referenceSearcher)
             PsiModifier.PUBLIC -> Visibility.PUBLIC
-
             else -> null
         }?.let {
             JKVisibilityModifierElement(it)
         }?.also { modifier ->
             assignNonCodeElements?.also { it(modifier, child) }
         }
-    }?.firstOrNull() ?: JKVisibilityModifierElement(Visibility.INTERNAL)
+    }?.firstOrNull()
+
+    if (visibility != null) return visibility
+    // for Kotlin classes, modifierList.children() returns null
```

Comment:
I think it's better to have a separate special method for Kotlin visibility, because it is only needed in one special case (computing visibility of Kotlin overridden methods), and it is kinda unexpected to have this code mixed in here.